pkg/binary/envoy/adminport.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package envoy
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"path/filepath"
+
+	"github.com/tetratelabs/log"
+
+	"github.com/tetratelabs/getenvoy/pkg/binary"
+)
+
+// EnableAdminAddressDetection sets the "--admin-address-path" flag and reads it back before attempting ready checks.
+//
+// Notably, this allows ephemeral admin ports via bootstrap configuration admin/port_value=0
+func EnableAdminAddressDetection(r *Runtime) {
```

Comment:
I owe unit tests on this, but waiting to see if this passes CI first as there are some files that only build on linux and tests that only execute on linux. Once this PR passes, I'll backfill tests on this..